envoyauth/evaluation.go

```diff
@@ -27,16 +28,16 @@ type EvalContext interface {
 	SetPreparedQuery(*rego.PreparedEvalQuery)
 }
 
-//Eval - Evaluates an input against a provided EvalContext and yields result
-func Eval(ctx context.Context, evalContext EvalContext, input ast.Value, result *EvalResult, opts ...func(*rego.Rego)) error {
+// Eval - Evaluates an input against a provided EvalContext and yields result
+func Eval(ctx context.Context, logger logging.Logger, evalContext EvalContext, input ast.Value, result *EvalResult, opts ...func(*rego.Rego)) error {
```

Comment:
side note: This is a good example of why we want to use structs or functional options for arguments... we can't extend this API without breaking callers.

Reply:
I guess if we have to break it, we could break it in a way that gets us into a better place...

Reply:
Alternatively you could have added the logger to EvalContext. I will update our usage after this is released.

Reply:
That's true, but not much less backward-compatibility-breaking, is it? Extending the EvalContext interface, I mean. But  I'll play with that a bit, and see if it ends up neater anyways.